classes/persistent/linkhistory.php

```diff
@@ -73,7 +73,7 @@ public static function delete_link_history($linkids) {
     public static function count_clicked_links($userid, $noticeid, $linkid = 0) {
         global $DB;
         $params = [];
-        if ($linkid > 0) {
+        if ($linkid > 0 && is_numeric($linkid)) {
```

Comment:
how about comparison reported in the issue? Should we also fix its logic?

Reply:
Actually I think the right spot for fixing it should be acknowledged_notice::other_cols. 
The current method should be fixed if it would have a type hint for $linkid. And it will get one at some point.

Reply:
I have created https://github.com/catalyst/moodle-local_sitenotice/issues/70